src/geo/data-providers/geojson/geojson-data-provider.js

```diff
@@ -15,33 +15,37 @@ var GeoJSONDataProvider = function (vectorLayerView, layerIndex) {
 // TODO: We can extract each "generator" to an individual file so that this file doesn't get too BIG
 GeoJSONDataProvider.prototype._dataGeneratorsForDataviews = {
   category: function (features, options) {
+    var filter = this._vectorLayerView.renderers[this._layerIndex].filter;
```

Comment:
As discussed offline, it would be nice if the vector layer provided a method to get the filter for a give layer so that we don't have to know so many internal details. Something like:

```
getFilter: function (layerIndex) {
   return this.renderers[layerIndex].filter;
}
```